fiona/_shim2.pyx

```diff
@@ -49,10 +51,12 @@ cdef void* gdal_open_vector(char* path_c, int mode, drivers, options):
 
     open_opts = CSLAddNameValue(open_opts, "VALIDATE_OPEN_OPTIONS", "NO")
 
-    flags = GDAL_OF_VECTOR | mode
     try:
         cogr_ds = GDALOpenEx(
             path_c, flags, <const char *const *>drvs, open_opts, NULL)
```

Comment:
@snorfalorpagus let's wrap the `GDALOpenEx` call in `exc_wrap_pointer()` https://github.com/Toblerity/Fiona/blob/master/fiona/_err.pyx#L222